### PR TITLE
[bitnami/logstash] Adding service account config for workload identity integration

### PR DESCRIPTION
**Pull request by luizferreira094**
…y integration

Signed-off-by: Luiz Ferreira <luiz.ferreira@idwall.co>

**Description of the change**

This config enables the logstash chart to integrate with workload identity of GCP, useful when you need to use a cloud permission on your config, like, buckets, pubsub, etc..

**Benefits**

This change implements a cleaner config of permissions when installing on a GKE cluster, with that you don't need to create a JSON Key for the logstash service acount with the cloud permission and put it inside the pod, you can config that easily using workload identity(https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity)

**Possible drawbacks**

This is a simply configuration, doesn't affect anything related to how things work right now.

**Applicable issues**

  - fixes #

**Additional information**

I created this PR because i was integrating this logstash chart on my cluster and i needed a special permission for using a custom plugin of pubsub(https://www.elastic.co/guide/en/logstash/current/plugins-inputs-google_pubsub.html), so, instead of creating a new service account on GCP project, download de jsonkey, and put it in a volume of pod, i created this config for easily set this permission using workload identity, using that, all i need to do is add an annotation on service account inside the cluster, pointing to the service account created on GCP project. 

**Checklist**

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the values.yaml and added to the README.md using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/logstash/templates/serviceaccount.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{{- if .Values.serviceAccount.create -}}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ include "logstash.serviceAccountName" . }}
```
Comment (carrodher) [inline bitnami/logstash/templates/serviceaccount.yaml:5]:
```suggestion
  name: {{ template "logstash.serviceAccountName" . }}
  namespace: {{ .Release.Namespace | quote }}
```

**Review by carrodher — COMMENTED** [review]
Thanks for your contribution, I just added some suggestions, can you please take a look?

File: bitnami/logstash/templates/serviceaccount.yaml (hunk 1 above)
Comment (luizferreira094) [reply bitnami/logstash/templates/serviceaccount.yaml]: done

File: bitnami/logstash/values.yaml [hunk 2]
```diff
@@ -162,6 +162,13 @@ extraVolumes: []
 ##     readOnly: true
 ##
 extraVolumeMounts: []
+
+## @param serviceAccount.create If true, create a service account object on Kubernetes.
+# serviceAccount:
+#   create: false
+serviceAccount: 
+  create: false
```
Comment (carrodher) [inline bitnami/logstash/values.yaml:170]: There are some values used in the template but not defined in the values, see
```suggestion
 ## ServiceAccount for Logstash
 ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/
 ##
 serviceAccount:
   ## @param serviceAccount.create Enable creation of ServiceAccount for Logstash pods
   ##
   create: true
   ## @param serviceAccount.name The name of the service account to use. If not set and `create` is `true`, a name is generated
   ## If not set and create is true, a name is generated using the logstash.serviceAccountName template
   ##
   name: ""
   ## @param serviceAccount.automountServiceAccountToken Allows automount of ServiceAccountToken on the serviceAccount created
   ## Can be set to false if pods using this serviceAccount do not need to use K8s API
   ##
   automountServiceAccountToken: true
   ## @param serviceAccount.annotations Additional custom annotations for the ServiceAccount
   ##
   annotations: {}
```

Comment (luizferreira094) [reply bitnami/logstash/values.yaml]: yeah you're right! thanks for the suggestion!

**Review by carrodher — DISMISSED** [review]
LGTM, thanks

**Pull request merged**